crypto/box/src/hkdf_box.rs

```diff
@@ -43,26 +44,28 @@ where
     KexAlgo: Kex,
     for<'privkey> <KexAlgo as Kex>::Public: From<&'privkey <KexAlgo as Kex>::EphemeralPrivate>,
     DigestAlgo: Digest + Update + FixedOutput + Default + Clone + BlockInput + Reset,
-    AeadAlgo: AeadInPlace + NewAead,
+    AeadAlgo: AeadInPlace + NewAead + CtAeadDecrypt,
     // Note: I think all of these bounds should go away after RFC 2089 is implemented
     // https://github.com/rust-lang/rfcs/blob/master/text/2089-implied-bounds.md
     <<KexAlgo as Kex>::Public as ReprBytes>::Size:
-        ArrayLength<u8> + Unsigned + Add<AeadAlgo::TagSize>,
-    Sum<<KexAlgo::Public as ReprBytes>::Size, AeadAlgo::TagSize>: ArrayLength<u8>,
+        ArrayLength<u8> + Unsigned + Add<<AeadAlgo as AeadInPlace>::TagSize>,
+    Sum<<KexAlgo::Public as ReprBytes>::Size, <AeadAlgo as AeadInPlace>::TagSize>: ArrayLength<u8>,
     GenericArray<u8, <<KexAlgo as Kex>::Public as ReprBytes>::Size>: Concat<
         u8,
-        AeadAlgo::TagSize,
-        Rest = GenericArray<u8, AeadAlgo::TagSize>,
+        <AeadAlgo as AeadInPlace>::TagSize,
+        Rest = GenericArray<u8, <AeadAlgo as AeadInPlace>::TagSize>,
         Output = GenericArray<
             u8,
-            <<<KexAlgo as Kex>::Public as ReprBytes>::Size as Add<AeadAlgo::TagSize>>::Output,
+            <<<KexAlgo as Kex>::Public as ReprBytes>::Size as Add<
+                <AeadAlgo as AeadInPlace>::TagSize,
+            >>::Output,
```

Comment:
yeah, i agree the concat stuff is messy... i can try to make a PR to kill it if that's helpful, or make a commit that goes into this PR for that

Reply:
I'm okay with it being messy, I'm just not as certain of my Rust skills in this area as I am with my C++ skills.

Reply:
With C++ I can visualize the memory layout and go from there but with Rust everything is sneaky...